File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/v2/ErrorJson.java
package ua.softgroup.matrix.server.supervisor.producer.json.v2;

import com.fasterxml.jackson.annotation.JsonView;
import ua.softgroup.matrix.server.supervisor.producer.json.JsonViewType;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class ErrorJson {

    @JsonView(JsonViewType.OUT.class)
    private String message;

    public ErrorJson(String message) {
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Override
    public String toString() {
        return "ErrorJson{" +
                "message='" + message + '\'' +
                '}';
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/v2/ErrorJson.scala
package ua.softgroup.matrix.server.supervisor.producer.json.v2

import com.fasterxml.jackson.annotation.JsonView
import ua.softgroup.matrix.server.supervisor.producer.json.JsonViewType

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class ErrorJson {

  @JsonView(Array(classOf[JsonViewType.OUT]))
  @BeanProperty
  var message: String = _

  def this(message: String) {
    this()
    this.message = message
  }

  override def toString = s"ErrorJson(message=$message)"

}
